### PR TITLE
Add Mail app with static templates and logging

### PR DESCRIPTION
**Pull request by weslord**

### CODE REVIEW

File: apps/mail/base.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            mail_url = reverse(
+                'admin:%s_%s_change' % ('mail', 'mail'),
+                args=[mail.id],
+            )
+            msg = format_html(
+                'The mail message "<a href ="{}">{}</a>" was generated successfully.',
+                mail_url,
+                mail,
+            )
+            messages.success(request, msg)
```
Comment (staydecent) [inline apps/mail/base.py]: How is this utilized? Would this show up in the django admin? Does DRF hook into the messages somehow?

File: apps/mail/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+```python
+class MailNewWheelbarrows(MailBase):
+    name = 'New Wheelbarrows'
+    subject = 'BarrowTrader | New wheelbarrow recommendations in the {{ city_name }} area!'
+    template = 'email/new_wheelbarrows.html
+
+    @classmethod
+    def process_context(cls, user, request, **kwargs):
+        ctx = {
+          customer_name = user.name,
```
[the remark this answers is not in the document]
Comment (weslord) [reply apps/mail/README.md]: 🤦‍♂

File: apps/mail/apps.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+from django.template.exceptions import TemplateDoesNotExist
+
+
+class MailConfig(AppConfig):
+    name = 'apps.mail'
+
+    def ready(self):
+
+        @register
+        def mailbase_check(app_configs, **kwargs):
```
[the remark this answers is not in the document]
Comment (weslord) [reply apps/mail/apps.py]: I modelled this after how Django's `check` validates the models, rather than the admin, but I can look into that pattern with the decorators if you think it's worthwhile.

This isn't required to get any of the functionality of the mail app, by the way, it's just so `manage.py check` will warn if any mail classes in `mail.py` are missing their templates.

File: apps/user/serializers.py [hunk 4]
```diff
@@ -24,6 +24,7 @@ def create(self, validated_data):
         user = get_user_model()(**validated_data)
         user.set_password(password)
         user.save()
+        MailWelcomeUser(user)
```
[the remark this answers is not in the document]
Comment (weslord) [reply apps/user/serializers.py]: `.send()` makes more sense.

File: apps/mail/base.py (hunk 1 above)
Comment (weslord) [reply apps/mail/base.py]: Yes, this is purely feedback at the top of the page for Django Admin actions - and a nice way to get a quick look at the mail you just triggered. (I found it pretty handy for development on Floorspace, testing trigger conditions and template updates)

I haven't messages having any effect on DRF (or workers), so I assume they're just discarded in those instances, but I'll look into that.

Might be worth adding a boolean flag to the class, `generate_admin_messages` or something, to disable these?

File: apps/mail/apps.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (staydecent) [reply apps/mail/apps.py]: Eh, it's not a major issue, especially if it's significantly more work. I rather get this merged sooner than later so we can stop using SendGrid templates in new projects.

Plus it was just a feeling back in September, the code works and isn't too hard to follow, 👍

File: apps/mail/base.py (hunk 1 above)
Comment (staydecent) [reply apps/mail/base.py]: Ah OK, that makes more sense. It's likely fine as is. I doubt DRF pays any attention to these. Boolean could be nice, but I'd say isn't a requirement.

File: apps/mail/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+### Create mail interface:
+
+Subclass `MailBase` and define the properties you want to customize.
+
+`apps/barrowtrader/mail.py`
+```python
+class MailNewWheelbarrows(MailBase):
+    name = 'New Wheelbarrows'
+    subject = 'BarrowTrader | New wheelbarrow recommendations in the {{ city_name }} area!'
+    template = 'email/new_wheelbarrows.html
```
Comment (staydecent) [inline apps/mail/README.md]: missing closing `'`

File: apps/mail/tasks.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        return
+
+    headers = {
+        'authorization': 'Bearer {0}'.format(settings.SENDGRID_API_KEY),
+        'content-type': 'application/json'
+    }
+
+    to_email = getattr(mail.user, mail.user.get_email_field_name())
+    try:
+        to_name = u.get_full_name()
```
Comment (staydecent) [inline apps/mail/tasks.py]: I think this should be `mail.user.get_full_name()`?

Comment (weslord) [reply apps/mail/tasks.py]: It's in a `try:` block so it's okay
